workflow/schemas/config.schema.yaml

```diff
@@ -56,100 +120,69 @@ properties:
-        description: parameters that should be forwarded to sorting (ie, number of threads)
+        description: parameters that should be forwarded
 
-  bwa_mem_merge:
+  samtools_index:
     type: object
+    description: parameters for indexing a bam file
     properties:
       benchmark_repeats:
         type: integer
-        description: set number of times benchmark should be repeated for bwa_mem_merge
+        description: set number of times benchmark should be repeated
       container:
         type: string
         description: name or path to docker/singularity container
       extra:
         type: string
-        description: parameters that should be forwarded to bwa_mem_merge
+        description: parameters that should be forwarded
 
-  samtools_sort:
+  samtools_merge_bam:
     type: object
+    description: parameters for merging  bam files
     properties:
       benchmark_repeats:
         type: integer
-        description: set number of times benchmark should be repeated for samtools_sort
+        description: set number of times benchmark should be repeated
       container:
         type: string
         description: name or path to docker/singularity container
       extra:
         type: string
-        description: parameters that should be forwarded to samtools_sort
-
-  extract_reads:
-    type: object
-    properties:
-      benchmark_repeats:
-        type: integer
-        description: set number of times benchmark should be repeated for extract_reads
-      container:
-        type: string
-        description: name or path to docker/singularity container
+        description: parameters that should be forwarded
 
-  mark_duplicates:
+  samtools_sort:
     type: object
+    description: parameters for sorting bam files
     properties:
       benchmark_repeats:
         type: integer
-        description: set number of times benchmark should be repeated for mark_duplicates
+        description: set number of times benchmark should be repeated
       container:
         type: string
         description: name or path to docker/singularity container
-
-  samtools_index:
-    type: object
-    properties:
-      benchmark_repeats:
-        type: integer
-        description: set number of times benchmark should be repeated for samtools_index
-      container:
+      extra:
         type: string
-        description: name or path to docker/singularity container
+        description: parameters that should be forwarded
 
-  merge_bam:
-    type: object
-    properties:
-      benchmark_repeats:
-        type: integer
-        description: set number of times benchmark should be repeated for merge_bam
-      container:
-        type: string
-        description: name or path to docker/singularity container
+  trimmer_software:
+    type: string
+    description: Set input files for alignment software, if "None" is set the files in units.tsv will be used. Default is None.
+    pattern: "^(fastp_pe|merged|None)$"
```

Comment:
Having `merged` as an option here is confusing as it should be a `trimmer_software`...

Reply:
Should we remove this completely if we aren't going to allow for starting from merged reads? I'm fine with that.

Reply:
People can still choose it in their workflows if they redefine the input files for `bwa_mem` but maybe we can just remove it here. But `None` needs to be handled still, right?

Reply:
@marrip  we should maybe rename it to fastq_source with the option of setting:
- raw to start from units file
- fastp_pe to start from trimmed, and have it as default?

Reply:
We could do that but I guess the beauty of `trimmer_software` is that it exists both in `prealignment` and `alignment` with the same input and we can handle trimmed reads by different trimmers as well as raw reads if `None` is specified. But I am open to alternatives.

Reply:
let's leave it as it is, for now at least

Reply:
👍